docs/docs/building-with-components.md

```diff
@@ -167,3 +167,35 @@ These are examples of the different ways React components are used in Gatsby
 sites. To see full working examples, check out the
 [examples directory](https://github.com/gatsbyjs/gatsby/tree/master/examples) in
 the Gatsby repo.
+
+### Querying data in components using StaticQuery
```

Comment:
I think this should be more about what "non-page components" are than how to use StaticQuery. The reason to write about it is to support StaticQuery docs, but this one should be focused more on the type of component that StaticQuery can be used in.